### PR TITLE
Restore plugin name for backwards compatibility

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**

### CODE REVIEW

File: command/auth_list.go [hunk 1]
```diff
@@ -155,18 +155,40 @@ func (c *AuthListCommand) detailedMounts(auths map[string]*api.AuthMount) []stri
...
-			mount.SealWrap,
-			mount.Options,
-			mount.Description,
-		))
+		pluginName := mount.Type
+		if pluginName == "plugin" {
+			pluginName = mount.Config.PluginName
+		}
+
+		if mount.Type == "plugin" {
```
Comment (briankassouf) [inline command/auth_list.go]: I don't think this branch is needed anymore, we should simply just replace mount.Type with pluginName

Comment (briankassouf) [reply command/auth_list.go]: Also above on line 146 we should s/Type/Plugin/ and remove the existing Plugin column

File: command/auth_list.go [hunk 2]
```diff
@@ -155,18 +155,40 @@ func (c *AuthListCommand) detailedMounts(auths map[string]*api.AuthMount) []stri
...
+				mount.Config.TokenType,
+				replication,
+				mount.SealWrap,
+				mount.Options,
+				mount.Description,
+			))
+		} else {
+			out = append(out, fmt.Sprintf("%s | %s | %s | %s | %s | %s | %s | %t | %v | %s",
+				path,
+				mount.Type,
```
Comment (briankassouf) [inline command/auth_list.go]:
```suggestion
				pluginName,
```

File: command/secrets_list.go [hunk 3]
```diff
@@ -155,18 +155,39 @@ func (c *SecretsListCommand) detailedMounts(mounts map[string]*api.MountOutput)
...
+				mount.Config.ForceNoCache,
+				replication,
+				mount.SealWrap,
+				mount.Options,
+				mount.Description,
+			))
+		} else {
+			out = append(out, fmt.Sprintf("%s | %s | %s | %s | %s | %t | %s | %t | %v | %s",
+				path,
+				mount.Type,
```
Comment (briankassouf) [inline command/secrets_list.go]:
```suggestion
				pluginName,
```

File: command/secrets_list.go [hunk 4]
```diff
@@ -155,18 +155,39 @@ func (c *SecretsListCommand) detailedMounts(mounts map[string]*api.MountOutput)
...
-			mount.SealWrap,
-			mount.Options,
-			mount.Description,
-		))
+		pluginName := mount.Type
+		if pluginName == "plugin" {
+			pluginName = mount.Config.PluginName
+		}
+
+		if mount.Type == "plugin" {
```
Comment (briankassouf) [inline command/secrets_list.go]: Same as above down here

File: api/sys_mounts.go [hunk 5]
```diff
@@ -133,6 +133,7 @@ type MountInput struct {
 	Description string            `json:"description"`
 	Config      MountConfigInput  `json:"config"`
 	Local       bool              `json:"local"`
+	PluginName  string            `json:"plugin_name,omitempty"`
```
Comment (calvn) [inline api/sys_mounts.go]: We should add a comment indicating this is a deprecated field (same for fields below).

Comment (tyrannosaurus-becks) [reply api/sys_mounts.go]: Addressed in https://github.com/hashicorp/vault/pull/5825/commits/a88a9c4654eb2e34841622c1e76cd974bb6e8fb5

File: command/auth_list.go [hunk 6]
```diff
@@ -143,7 +143,7 @@ func (c *AuthListCommand) detailedMounts(auths map[string]*api.AuthMount) []stri
 		}
 	}
 
-	out := []string{"Path | Type | Accessor | Plugin | Default TTL | Max TTL | Token Type | Replication | Seal Wrap | Options | Description"}
+	out := []string{"Path | Plugin | Accessor | Default TTL | Max TTL | Token Type | Replication | Seal Wrap | Options | Description"}
```
Comment (calvn) [inline command/auth_list.go]: Should it be "Plugin" or "Type"?

Comment (tyrannosaurus-becks) [reply command/auth_list.go]: Brian had commented `s/Type/Plugin/`. I can go the other way if that's desired, but we should all agree.

Comment (briankassouf) [reply command/auth_list.go]: I think it's fine to just say "Plugin" now since everything is a plugin

Comment (calvn) [reply command/auth_list.go]: 👍It might look weird listing older servers since that's not the case, but probably not a big deal.

**Pull request merged**